### ISSUE TITLE
Gradle Plugin: Cannot find version resource when another plugin has a version.properties

### ISSUE THREAD

**Issue by timothyolt** [detekt/detekt#4830]
## Expected Behavior
When another Gradle plugin that has a `version.properties` resource is added to the same project as Detekt,
Detekt should still be able to find it's version.

## Observed Behavior
Instead, the Gradle configuration crashes with an exception:
```
org.gradle.api.internal.plugins.PluginApplicationException: Failed to apply plugin 'io.gitlab.arturbosch.detekt'.
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:173)
        at org.gradle.api.internal.plugins.DefaultPluginManager.apply(DefaultPluginManager.java:146)
        at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.lambda$applyLegacyPlugin$2(DefaultPluginRequestApplicator.java:160)
        at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.applyPlugin(DefaultPluginRequestApplicator.java:201)
        ... 155 more
Caused by: org.gradle.api.reflect.ObjectInstantiationException: Could not create an instance of type io.gitlab.arturbosch.detekt.extensions.DetektExtension.
        at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.doCreate(DependencyInjectingInstantiator.java:69)
        at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.newInstanceWithDisplayName(DependencyInjectingInstantiator.java:50)
        at org.gradle.internal.extensibility.DefaultConvention.instantiate(DefaultConvention.java:216)
        at org.gradle.internal.extensibility.DefaultConvention.create(DefaultConvention.java:125)
        at io.gitlab.arturbosch.detekt.DetektPlugin.apply(DetektPlugin.kt:17)
        at io.gitlab.arturbosch.detekt.DetektPlugin.apply(DetektPlugin.kt:13)
        at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:43)
        at org.gradle.api.internal.plugins.RuleBasedPluginTarget.applyImperative(RuleBasedPluginTarget.java:51)
        at org.gradle.api.internal.plugins.DefaultPluginManager.addPlugin(DefaultPluginManager.java:187)
        at org.gradle.api.internal.plugins.DefaultPluginManager.access$100(DefaultPluginManager.java:52)
        at org.gradle.api.internal.plugins.DefaultPluginManager$AddPluginBuildOperation.run(DefaultPluginManager.java:282)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
        at org.gradle.api.internal.plugins.DefaultPluginManager.lambda$doApply$0(DefaultPluginManager.java:167)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:44)
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:166)
        ... 158 more
Caused by: java.lang.NullPointerException: Properties().run {
     …ty("detektVersion")
    } must not be null
        at io.gitlab.arturbosch.detekt.extensions.DetektExtensionKt.loadDetektVersion(DetektExtension.kt:111)
        at io.gitlab.arturbosch.detekt.extensions.DetektExtension.<init>(DetektExtension.kt:14)
        at io.gitlab.arturbosch.detekt.extensions.DetektExtension_Decorated.<init>(Unknown Source)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at org.gradle.internal.instantiation.generator.AsmBackedClassGenerator$InvokeConstructorStrategy.newInstance(AsmBackedClassGenerator.java:2078)
        at org.gradle.internal.instantiation.generator.AbstractClassGenerator$GeneratedClassImpl$GeneratedConstructorImpl.newInstance(AbstractClassGenerator.java:488)
        at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.doCreate(DependencyInjectingInstantiator.java:64)
        ... 179 more
```

## Steps to Reproduce
The easiest way to reproduce is to add both Gradle plugins to the same project.
```
plugins {
    id("io.github.detekt.gradle.compiler-plugin")
    id("io.gitlab.arturbosch.detekt")
}
```
Though this will happen with any other plugin that has a `versions.properties`
You could also make a new empty Gradle plugin with an empty `versions.properties` and get the same effect.

## Your Environment

* Version of detekt used: 1.20.0
* Version of compiler plugin: [main 770c973bd983722fe9cfe5d539d71718f9ddf8b1](https://github.com/detekt/detekt-compiler-plugin/tree/770c973bd983722fe9cfe5d539d71718f9ddf8b1)
* Version of Gradle used (if applicable): 7.4.2
* Gradle scan link (add `--scan` option when running the gradle task): https://scans.gradle.com/s/hfodp3pq3znwg
* Operating System and version: MacOS 11.6.1

**Comment by cortinico** [comment]
Closing as this duplicates https://github.com/detekt/detekt-intellij-plugin/issues/161

**Comment by cortinico** [comment]
Sorry I accidentally closed this issue instead of another

**Comment by BraisGabin** [comment]
Fixed by #4830
